lib/csu/common-cheri/crt_init_globals.c

```diff
@@ -77,22 +77,23 @@ crt_init_globals(void)
 #ifndef POSITION_INDEPENDENT_STARTUP
 /* This is __always_inline since it is called before globals have been set up */
 static __always_inline void
-do_crt_init_globals(const Elf_Phdr * __capability phdr, long phnum)
+do_crt_init_globals(const Elf_Phdr *phdr, long phnum)
 {
-	const Elf_Phdr * __capability phlimit = phdr + phnum;
+	const Elf_Phdr *phlimit = phdr + phnum;
 	Elf_Addr text_start = (Elf_Addr)-1l;
 	Elf_Addr text_end = 0;
 	Elf_Addr readonly_start = (Elf_Addr)-1l;
 	Elf_Addr readonly_end = 0;
 	Elf_Addr writable_start = (Elf_Addr)-1l;
 	Elf_Addr writable_end = 0;
-
 	bool have_rodata_segment = false;
 	bool have_text_segment = false;
 	bool have_data_segment = false;
+	void * __capability code_cap, * __capability data_cap,
+	     * __capability rodata_cap;
```

Comment:
```suggestion
	const void * __capability code_cap;
	const void * __capability data_cap;
	const void * __capability rodata_cap;
```
Multiple-variable declaration that aren't simple types such as int are error-prone and not particularly readable. Also you are dropping const from the variable.

Reply:
error: variable 'rodata_cap' is uninitialized when used here [-Werror,-Wuninitialized]
                rodata_cap = cheri_setaddress(rodata_cap, readonly_start);

Reply:
Dropping `const` was deliberate, it's a stupid thing to have on these.

Reply:
it indicates that it's only used to derive non-writable capabilities.

Reply:
Oh, hm, they are `const` in `cheri_init_globals_3`, ok, will revert that part of things.